x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh

```diff
@@ -2,27 +2,120 @@
 
 API_KEY_ENCODED=$1
 API_ENDPOINT=$2
+AUTO_DOWNLOAD_CONFIG=$3
 
 updateStepProgress() {
-  echo "  GET $API_ENDPOINT/step/$1?status=$2"
+  local STEPNAME="$1"
+  local STATUS="$2" # "incomplete" | "complete" | "disabled" | "loading" | "warning" | "danger" | "current"
   curl --request GET \
-    --url "$API_ENDPOINT/step/$1?status=$2 2023-05-24" \
-    --header "Authorization: ApiKey $API_KEY_ENCODED" \
+    --url "${API_ENDPOINT}/custom_logs/step/${STEPNAME}?status=${STATUS}" \
+    --header "Authorization: ApiKey ${API_KEY_ENCODED}" \
     --header "Content-Type: application/json" \
-    --header "kbn-xsrf: true"
-  echo ""
+    --header "kbn-xsrf: true" \
+    --output /dev/null \
+    --no-progress-meter
 }
 
-echo "Downloading Elastic Agent"
+echo "Downloading Elastic Agent archive"
 # https://www.elastic.co/guide/en/fleet/8.7/install-standalone-elastic-agent.html
-curl -L -O https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.7.1-linux-x86_64.tar.gz
-updateStepProgress "ea-download" "success"
+updateStepProgress "ea-download" "loading"
+curl -L -O https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.7.1-linux-x86_64.tar.gz --fail --continue-at -
+if [ "$?" -eq 0 ]; then
+  echo "Downloaded Elastic Agent"
+  updateStepProgress "ea-download" "complete"
+else
+  updateStepProgress "ea-download" "danger"
+  echo "Failed to download Elastic Agent"
+  exit 1
+fi
+
 echo "Extracting Elastic Agent"
-tar xzvf elastic-agent-8.7.1-linux-x86_64.tar.gz
-updateStepProgress "ea-extract" "success"
+updateStepProgress "ea-extract" "loading"
+tar -xzf elastic-agent-8.7.1-linux-x86_64.tar.gz --checkpoint=.1000
```

Comment:
shouldn't the version of the elastic agent be included as param? in this way we could get the latest one without changing the script

Reply:
yes it needs to be parameterized, I can push an update for this